omise/controllers/front/payment.php

```diff
@@ -0,0 +1,28 @@
+<?php
+if (! defined('_PS_VERSION_')) {
+    exit();
+}
+
+if (defined('_PS_MODULE_DIR_')) {
+    require_once _PS_MODULE_DIR_ . '/omise/classes/charge.php';
+}
+
```

Comment:
@nimid if charge is a controller, then it's no reason to have `flow controller` as `OmisePaymentModuleFrontController` isn't it?

If you care about "Along with the rise of the features that will be developed soon, the class OmisePaymentModuleFrontController will has more size, logic and complexity that adversely affect to the maintenance. For an example from another project, function process_payment()."

Then, just have more controller with multiple endpoints.

i.e.
- controllers/charge.php
- controllers/refund.php
- controllers/threedsecure.php
- controllers/internetbanking.php

I don't see the benefit of have 2 controllers (delegated class & flow controller class).

Say, if you have to develop more feature in the future, your `charge.php` will "has more size, logic and complexity that adversely affect to the maintenance" anyway, since that `flow controller class` do nothing, just bypass action to another delegated class.

---

Btw, to reduce confusing, if `charge.php` is a controller, could you move it to `controllers` folder?

---

"If it has calling, both static call and extending are the dependency."
Yup, we have no way to retrieve "orderTotal()" if we are not using `Cart` dependency.
But the question is, if `ModuleFrontController` already provided by itself, why don't we just use `$this->context->cart` from itself?

👆 btw, this one may be a totally different topic, I have this idea because at first I confused and thought that `classes/charge.php` is a model-like class.